### PR TITLE
 samples: bluetooth: Enable mesh_demo on mimxrt10{50,60,64}_evk boards

### PR DESCRIPTION
**Pull request by MaureenHelm**
Uses lvgl to implement board.h hooks for lcd-enabled imx rt boards. In
this configuration, the application briefly displays a popup window on
the lcd when a bluetooth mesh message is received.

Note that these boards require a frdm_kw41z shield for bluetooth
controller support.

This also works on the reel_board (with an additional `prj_reel_board.conf` not currently included), but the display update is so slow that I'm not sure it makes sense to advertise it.

### CODE REVIEW

File: samples/bluetooth/mesh_demo/README.rst [hunk 1]
```diff
@@ -44,6 +44,16 @@ the Zephyr tree.
 See :ref:`bluetooth samples section <bluetooth-samples>` for details on how
 to run the sample inside QEMU.
 
+For mimxrt10xx_evk boards with an lcd and frdm_kw41z shield, build and flash
```
Comment (dbkinder) [inline samples/bluetooth/mesh_demo/README.rst]:
```suggestion
For mimxrt10xx_evk boards with an LCD and a frdm_kw41z shield, build and flash
```

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Review by jhedberg — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Review by vanwinkeljan — APPROVED** [review]
+1 on the lvgl code

**Comment by MaureenHelm** [comment]
@jhe shippable is failing because of a build warning:
```
/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/subsys/bluetooth/host/mesh/net.c:57: error: "SRC" redefined [-Werror]
 #define SRC(pdu)           (sys_get_be16(&(pdu)[5]))
 
In file included from ../../../../../../ext/hal/nxp/mcux/devices/MIMXRT1062/fsl_device_registers.h:23,
                 from ../../../../../../ext/hal/nxp/mcux/drivers/imx/./fsl_common.h:22,
                 from ../../../../../../soc/arm/nxp_imx/rt/soc.h:18,
                 from ../../../../../../include/arch/arm/cortex_m/cmsis.h:21,
                 from ../../../../../../include/arch/arm/cortex_m/mpu/arm_mpu_v7m.h:10,
                 from ../../../../../../include/arch/arm/cortex_m/mpu/arm_mpu.h:13,
                 from ../../../../../../include/arch/arm/arch.h:182,
                 from ../../../../../../include/arch/cpu.h:17,
                 from ../../../../../../include/kernel_includes.h:34,
                 from ../../../../../../include/kernel.h:17,
                 from ../../../../../../include/zephyr.h:18,
                 from /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/subsys/bluetooth/host/mesh/net.c:9:
../../../../../../ext/hal/nxp/mcux/devices/MIMXRT1062/MIMXRT1062.h:35508: note: this is the location of the previous definition
 #define SRC                                      ((SRC_Type *)SRC_BASE)
```

The `SRC(pdu)` macro is only used once. How do you feel about removing or renaming it?

File: samples/bluetooth/mesh_demo/README.rst (hunk 1 above)
Comment (MaureenHelm) [reply samples/bluetooth/mesh_demo/README.rst]: Thanks, fixed.

**Review by dbkinder — APPROVED** [review]

**Review by jhqian — APPROVED** [review]

**Comment by MaureenHelm** [comment]
> The `SRC(pdu)` macro is only used once. How do you feel about removing or renaming it?

@jhe ping.

**Comment by jhedberg** [comment]
>> The SRC(pdu) macro is only used once. How do you feel about removing or renaming it?
> @jhe ping.

@MaureenHelm sorry, I had missed your earlier comment. These are private macros (only defined within a single c-file) - wouldn't it make more sense to fix the public header file that's using such a short and generic name for a macro? The names that `mesh/net.c` uses are directly taken from the Mesh spec (see attached image), and they're not violating anyones namespace, so I'm slightly reluctant to having to change them :)

Screenshot 2019-04-19 at 15 31 18

**Comment by MaureenHelm** [comment]
Closing for now, until I figure out how to fix the build warnings.

**Pull request closed without merging**